test_tsp.py

```diff
@@ -0,0 +1,85 @@
+# The MIT License (MIT)
+#
+# Copyright (C) 2021 - Ericsson
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from tsp.tsp_client import TspClient
+import os
+import pytest
+
+
+class TestTspClient:
+    # Not a pytest fixture so that VS Code may find its definitions.
+    tsp_client = TspClient('http://localhost:8080/tsp/api/')
+
+    @pytest.fixture(scope='module')
+    def traces_path(self):
+        yield f'{os.getcwd()}/tracecompass-test-traces/ctf/src/main/resources/kernel'
+
+    def test_fetch_traces_none(self):
+        response = self.tsp_client.fetch_traces()
+        assert response.status_code == 200
+        assert not response.model.traces
+
+    def test_open_trace_twice_then_delete(self, traces_path):
+        response = self.tsp_client.open_trace('channel0_0', traces_path)
+        assert response.status_code == 200
+        response = self.tsp_client.open_trace('channel0_1', traces_path)
+        assert response.status_code == 200
+
+        response = self.tsp_client.fetch_traces()
+        assert len(response.model.traces) == 2
+        # Delete each previously opened trace to clean server for next run.
+        for trace in response.model.traces:
+            # Do not also delete the trace from disk; file part of this repo.
+            response = self.tsp_client.delete_trace(trace.UUID, False)
+            assert response.status_code == 200
+
+    def test_fetch_opened_trace_then_delete(self, traces_path):
+        response = self.tsp_client.open_trace('channel0_0', traces_path)
```

Comment:
One thing, it's kinda the convention to open the 'metadata' file. They all work though

Reply:
> One thing, it's kinda the convention to open the 'metadata' file. They all work though

Thanks. I need to look deeper into this, as I'm not familiar with this subtlety yet.

Reply:
-Done. I amended a few commits to add a second test trace path fixture, also fixing the opened trace names accordingly. Now each test trace path (namely kernel's and kernel_vm's) should properly be opened. Both paths or file contents are now treated as an actual trace each.

Thanks again and here is the diff for review: https://github.com/theia-ide/tsp-python-client/compare/f56af7e7fbd3e17b7c21f29a40da78a169f06ef5..170c551ef0662a361e6a9886e28333647e7ce964